src/app/proposals/components/LatestProposalsTable.tsx

```diff
@@ -51,25 +51,37 @@ const LatestProposalsTable = ({ proposals }: LatestProposalsTableProps) => {
   // State for proposal quick filters
   const [activeFilter, setActiveFilter] = useState<string | null>(null)
 
+  const clearSearchRef = useRef<() => void>()
+
   const handleFilterToggle = useCallback(
     (keyword: string) => () => {
       if (activeFilter === keyword) {
         // Reset both states if clicking active filter
         setActiveFilter(null)
         setSearchedProposal('')
+        clearSearchRef.current?.()
       } else {
         // Set new filter if clicking different filter
         setActiveFilter(keyword)
         setSearchedProposal(keyword)
+        clearSearchRef.current?.()
       }
     },
     [activeFilter],
   )
 
-  const handleSearch = useCallback((value: string) => {
-    setSearchedProposal(value)
-    setActiveFilter(null)
-  }, [])
+  const handleSearch = useCallback(
+    (value: string) => {
+      if (activeFilter) {
+        // If there's an active filter, only clear input without changing search state
+        clearSearchRef.current?.()
+        return
```

Comment:
What's the deal with using `if-else`? 😅

Reply:
Without this new addition, there's the edge case where we want to erase the text in the search input when a user selects a filter. Thanks for catching this other unintended side effect. I'm attempting to use a fix that doesn't introduce a new state